doc/x-interface.md

```diff
@@ -0,0 +1,105 @@
+      The adapter asserts `k_accept` and `k_writeback` accordingly.
+    - The instruction is not accepted by any of the connected predecoders.
+      The adapter de-asserts `k_accept` and `k_writeback` to indicate an illegal instruction has been encountered.
+- When both `q_valid` and `q_ready` are high, the transaction is successful
+
+### Response channel
+The response channel signals are:
+
+| Signal Name   | Range                   | Description                          |
+| ------------- | ----------------------- | ------------------------------------ |
+| `p_rd`        | `4:0`                   | Destination Register Address         |
+| `p_data0`     | `DataWidth-1:0`         | Primary Writeback Data               |
+| `p_data1`     | `DataWidth-1:0`         | Secondary Writeback Data             |
+| `p_dualwb`    | `0:0`                   | Dual-Writeback Response              |
+| `p_error`     | `0:0`                   | Error Flag                           |
+
+Additionally, the handshake signals `q_ready` and `q_valid` are implemented.
+
+Notes:
+  - `p_data0` and `p_data1` carry the response data resulting from offloaded instructions.
+    `p_data0` carries the default write-back data and is written to the destination register identified by `p_rd_id`.
+    `p_data1` is used only for dual-writeback instructions.
+  - Dual write-back instructions are marked by the accelerator sub-system by setting `p_dualwb`.
+  - The error flag included in the response channel indicates processing errors encountered by the accelerator.
+    The actions to be taken by a core to recover from accelerator errors are not yet defined.
+
+#### Response Transaction
+The response channel is handshaked according to the following scheme:
+- The initiator asserts `p_valid`. The assertion of `p_valid` must not depend on `p_ready`. The assertion of ready may depend on `p_valid`.
+- Once `p_valid` has been asserted all data must remain stable.
+- The receiver asserts `p_ready` whenever it is ready to receive the transaction. Asserting `p_ready` by default is allowed. While `p_valid` is low, `p_ready` may be retracted at any time.
+- When both `p_valid` and `p_ready` are high the transaction is successful.
+
+## Operand Origin
+The operands forwarded to the accelerator interconnect are determined in the same way as any regular RISC-V instructions.
+Any source registers from the integer register file of the offloading core as defined in the [RISC-V User-Level ISA specification](https://riscv.org/wp-content/uploads/2017/05/riscv-spec-v2.2.pdf#page=24) are allowed.
+If source registers are used, operand A, B and C contain `rs1`, `rs2` and `rs3` respectively.
+For ternary instructions, the instruction format R4-type instruction format is to be used, defining the `rs3` register address by bits `instr_data[31:27]`.
+
+## Write-back Destination
```

Comment:
nit: I think you are a bit inconsistent with write-back vs writeback (at least looking at the dual-writeback down there :-)).